### PR TITLE
Update AITER commit

### PR DESCRIPTION
**Pull request by Micky774**
# Description

This PR bumps our AITER submodule commit. TE builds successfully, and I am running local tests but wanted to establish this PR to allow CI a chance to run as well.

Fixes https://github.com/ROCm/frameworks-internal/issues/13466

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes

Please list the changes introduced in this PR:

- Bump AITER submodule commit

# Checklist:

- [ ] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [ ] The functionality is complete
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by wangye805 — CHANGES_REQUESTED** [review]
According to our discussion with AITER team, their fix should also fix the softmax-lse stride issue in MI350. Can you try to enable varlen bwd in MI350 (https://github.com/ROCm/TransformerEngine/blob/366f6282813baef373de6045a8cbef4ded6fce59/transformer_engine/common/fused_attn_rocm/fused_attn_ck.cpp#L1048-L1057) to see whether our pytests can pass now?

**Comment by Micky774** [comment]
@wangye805 I've updated the V3 dispatch to no longer disable certain cases, and all tests are passing locally. Once CI looks good, can we merge? Is there anything else you'd like me to check out or test here?

Edit: Actually I'm seeing some errors on MI350 -- let me investigate

**Review by wangye805 — APPROVED** [review]

**Comment by Micky774** [comment]
There are three main types of errors currently, all from `tests/pytorch/fused_attn/test_paged_attn.py`:
- Accuracy errors for `FlashAttention` using `bf16`.
  - This occurs for THD, BSHD, SBHD.
  - Config `test_paged_attn[False-False-TransformerLayer-FlashAttention-<True, False>-<arbitrary qkv-layout>-infer_0-dtype1]`
- A runtime error: `TransformerEngine/transformer_engine/common/fused_attn_rocm/utils_hip.cpp:220 in function generateMatrixStrides: Unsupported layout: 23` which also occurs with layouts `19, 21`
  - The layouts correspond to `paged_kv_bshd_bshd_bshd, paged_kv_thd_bshd_bshd, paged_kv_sbhd_bshd_bshd`
  - This seems like something we should be able to solve. Actually, I'm not sure why it was previously working...
- A runtime error: `TransformerEngine/transformer_engine/common/fused_attn_rocm/fused_attn_hip.cpp:771 in function nvte_fused_attn_fwd: Invalid combination of data type and sequence length for rocm fused attention.`
  - This occurs for `thd_bshd_bshd`

Edit: nvm we don't run/support `tests/pytorch/fused_attn/test_paged_attn.py` so this is a moot point for now.
